### PR TITLE
Add support for restricting packages to particular named parsers (A)

### PR DESCRIPTION
**Pull request by dpvc**
This PR adds a parser name to each configuration (default is `tex`) and uses an array of names to indicate which packages are allowed for the given `ParserConfiguration` instance.  This allows the `text-base` configuration to be marked as for the text parser, so it will be rejected if included in the `tex.packages` list.

Pros:  This is easy to implement, and allows mixing and matching for packages with parsers, even if the package's parser is not a superclass of the given parser.

Cons:  You need to list all the superclasses explicitly if you want them to be allowed.  So to allow configuring the `TextParser` to accept TeX packages, you must indicate `['tex', 'text']` when creating the `ParserConfiguration` for the `TextParser`.

This is one proposal for how to handle this.

### CODE REVIEW

**Review by zorkow — APPROVED** [review]
I prefer this option.

File: ts/input/tex/Configuration.ts [hunk 1]
```diff
@@ -284,13 +296,14 @@ export class ParserConfiguration {
    */
   public nodes: {[key: string]: any}  = {};
 
-
   /**
    * @constructor
    * @param {(string|[string,number])[]} packages A list of packages with
    *     optional priorities.
+   * @parm {string[]} parsers   The names of the parsers whose packages are supported
```
Comment (zorkow) [inline ts/input/tex/Configuration.ts:303]: Maybe rephrase a bit.

**Review by zorkow — APPROVED** [review]
I prefer this option.

File: ts/input/tex/Configuration.ts (hunk 1 above)
Comment (dpvc) [reply ts/input/tex/Configuration.ts]: Done
